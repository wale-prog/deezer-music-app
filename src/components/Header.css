.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 24px;
}

.mic {
  margin-right: 20px;
}

.header-main {
  background-color: #4369b2;
  color: #fff;
}

.tracks {
  display: flex;
  flex-direction: column;
}

.tracks p {
  margin: 0 0 2px 24px;
  padding: 0;
}

.mid-head {
  display: flex;
  justify-content: space-between;
}

.deezer-text {
  font-family: 'Dancing Script', cursive;
  font-weight: 500;
}

.deezer-logo {
  display: flex;
  flex-direction: column;
  font-size: 14px;
}

.deezer-logo p {
  margin: 0;
}
