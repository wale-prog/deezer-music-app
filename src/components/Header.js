import { useSelector } from 'react-redux';
import { GiHamburgerMenu as Menu } from 'react-icons/gi';
import { BsMic, BsSearch } from 'react-icons/bs';
import logo from '../images/Colored_Full_White.png';
import './Header.css';

const Header = () => {
  const charts = useSelector((state) => state.charts);

  return (
    <div className="header-main">
      <div className="header-s">
        <div className="header">
          <Menu />
          <p>All time charts</p>
          <div style={{ visibility: 'hidden' }}>
            <BsMic className="mic" />
            <BsSearch />
          </div>
        </div>
        <div className="mid-head">
          <div className="tracks">
            <p>
              {charts.length}
              {' '}
              Tracks
            </p>
          </div>
          <div className="deezer-logo">
            <p className="deezer-text">Powered by</p>
            <img style={{ width: '100px', marginRight: '24px' }} src={logo} alt="Deezer logo" />
          </div>
        </div>
      </div>

    </div>
  );
};
export default Header;
