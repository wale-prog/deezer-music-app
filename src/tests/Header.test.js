import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/index';
import Header from '../components/Header';

it('Header is rendered', () => {
  const detailPage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(detailPage).toMatchSnapshot();
});
