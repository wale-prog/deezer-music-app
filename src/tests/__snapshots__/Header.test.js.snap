// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered 1`] = `
<div
  className="header-main"
>
  <div
    className="header"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
      />
    </svg>
    <p>
      all time charts
    </p>
    <div>
      <svg
        className="mic"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
        <path
          d="M10 8a2 2 0 1 1-4 0V3a2 2 0 1 1 4 0v5zM8 0a3 3 0 0 0-3 3v5a3 3 0 0 0 6 0V3a3 3 0 0 0-3-3z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
        />
      </svg>
    </div>
  </div>
  <div
    className="mid-head"
  >
    <div
      className="tracks"
    >
      <p>
        0
      </p>
      <p>
        Tracks
      </p>
    </div>
    <div
      className="deezer-logo"
    >
      <p
        className="deezer-text"
      >
        Powered by
      </p>
      <img
        alt="Deezer logo"
        src="Colored_Full.png"
        style={
          Object {
            "marginRight": "24px",
            "width": "100px",
          }
        }
      />
    </div>
  </div>
</div>
`;
